pkg/core/state/notification_event.go

```diff
@@ -21,9 +21,11 @@ type NotificationEvent struct {
 }
 
 type ContractInvocation struct {
-	Hash      util.Uint160     `json:"contract_hash"`
-	Method    string           `json:"method"`
-	Arguments *stackitem.Array `json:"arguments"`
+	Hash           util.Uint160     `json:"contract_hash"`
+	Method         string           `json:"method"`
+	Arguments      *stackitem.Array `json:"arguments"`
+	ArgumentsCount uint32           `json:"arguments_count"`
```

Comment:
Why this one? In the optimistic case you have `Arguments` with some proper number of elements. In pessimistic, does it make any difference?

Reply:
The idea was that in case of `Isvalid` = false, then at least we have some information on the argument count.

as said in https://github.com/nspcc-dev/neo-go/pull/3569#discussion_r1743213122
> I don't think we should throw away the complete invocation record because of a parameter violation. If it's recorded on chain then it was apparently still a valid invocation, regardless if we want to store all of the invocation details

I don't recall a hard necessity for this in my original use-case for this PR, but I think it's cheap to do and harder to add later (assuming the C# follows).